src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -233,6 +245,47 @@ static Symbol getAttributeTarget(Binder current)
 
             static WithTypeParametersBinder getExtraWithTypeParametersBinder(Binder enclosing, Symbol target)
                 => target.Kind == SymbolKind.Method ? new WithMethodTypeParametersBinder((MethodSymbol)target, enclosing) : null;
+
+            // We're bringing parameters in scope inside `nameof` in attributes on methods, their type parameters and parameters.
+            // This also applies to local functions, lambdas, indexers and delegates.
+            static InMethodBinder getExtraInMethodBinder(Binder enclosing, Symbol target)
+            {
+                var method = target switch
+                {
+                    MethodSymbol methodSymbol => methodSymbol,
+                    ParameterSymbol parameter => getMethodFromParameter(parameter),
+                    TypeParameterSymbol typeParameter => getMethodFromTypeParameter(typeParameter),
+                    PropertySymbol property => property.GetMethod,
```

Comment:
>property.GetMethod [](http://example.com/codeflow?start=47&length=18)

This looks suspicious. I assume we are trying to make property parameters visible, but it looks like we are actually doing that for some other parameters and that is likely to be observable through GetSymbolInfo and the like APIs.